components/graph/git-node.js

```diff
@@ -175,38 +185,31 @@ class GitNodeViewModel extends Animateable {
   showRefSearchForm(obj, event) {
     this.refSearchFormVisible(true);
 
-    const textBox = event.currentTarget.parentElement.querySelector('input[type="search"]');
-    const $textBox = $(textBox);
-
-    if (!$textBox.autocomplete('instance')) {
-      const renderItem = (ul, item) => $(`<li><a>${item.displayHtml()}</a></li>`).appendTo(ul);
-      $textBox.autocomplete({
-        classes: {
-          'ui-autocomplete': 'dropdown-menu',
-        },
+    const textBox = event.currentTarget.nextElementSibling.firstElementChild; // this may not be the best idea...
+    $(textBox)
+      .autocomplete({
         source: this.refs().filter((ref) => !ref.isHEAD),
         minLength: 0,
-        create: (event) => {
-          $(event.target).data('ui-autocomplete')._renderItem = renderItem;
-        },
-        select: (_event, ui) => {
+        select: (event, ui) => {
           const ref = ui.item;
           const ray = ref.isTag ? this.tagsToDisplay : this.branchesToDisplay;
 
           // if ref is in display, remove it, else remove last in array.
           ray.splice(ray.indexOf(ref), 1);
           ray.unshift(ref);
           this.refSearchFormVisible(false);
-
-          // Clear search input on selection
-          return false;
         },
-      });
-      $textBox.focus((event) => {
-        $(event.target).autocomplete('search', event.target.value);
-      });
-      $textBox.autocomplete('search', '');
```

Comment:
this code seems unrelated to optimization, not sure if this is an improvement or an accidental revert